srcpkgs/boost/patches/align.patch

```diff
@@ -0,0 +1,28 @@
+This code is wrong and uses feature test macros to determine whether something
+is available; FTMs shouldn't be in user code outside of top level #define's.
+
+This lead to an issue on musl where libtorrent-rasterbar was including this
+header (via boost asio) from two different places, and segfaulting due to
+mismatched implementations -- one's the posix one using posix_memalign, the
+other one's the generic one that stores a pointer to itself.
+
+posix_memalign is always available on the libcs we support and should always be
+used, so we force that. We still leave applications with the option of forcing
+boost specific behavior, if they really want it.
+
+diff --git a/boost/align/aligned_alloc.hpp b/boost/align/aligned_alloc.hpp
+index 1d81a13..f2b5137 100644
+--- a/boost/align/aligned_alloc.hpp
++++ b/boost/align/aligned_alloc.hpp
+@@ -38,10 +38,8 @@ Distributed under the Boost Software License, Version 1.0.
+ #include <boost/align/detail/aligned_alloc_posix.hpp>
+ #elif defined(sun) || defined(__sun)
+ #include <boost/align/detail/aligned_alloc_sunos.hpp>
+-#elif (_POSIX_C_SOURCE >= 200112L) || (_XOPEN_SOURCE >= 600)
+-#include <boost/align/detail/aligned_alloc_posix.hpp>
+ #else
+-#include <boost/align/detail/aligned_alloc.hpp>
```

Comment:
We may need to:
```cpp
#ifndef _POSIX_C_SOURCE
#define _POSIX_C_SOURCE 200112L
#endif
```

Just in case the `boost/align/detail/aligned_alloc_posix.hpp` needs something only available when that macro is defined?